includes/data-events/connectors/class-mailchimp.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Newspack Data Events Mailchimp Connector
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Data_Events\Connectors;
+
+use \Newspack\Data_Events;
+use \Newspack\Mailchimp_API;
+use \Newspack\Newspack_Newsletters;
+use \Newspack\Reader_Activation;
+use \Newspack\WooCommerce_Connection;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main Class.
+ */
+class Mailchimp {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( Reader_Activation::is_enabled() && true === Reader_Activation::get_setting( 'sync_esp' ) ) {
+			Data_Events::register_handler( [ __CLASS__, 'reader_registered' ], 'reader_registered' );
+			Data_Events::register_handler( [ __CLASS__, 'donation_new' ], 'donation_new' );
```

Comment:
The methods other than these two seem to be strictly related to the API, I was surprised to find them here and not in `class-mailchimp-api.php`.

Reply:
I don't find the connector methods to be abstract enough to be used within the generic `Mailchimp_API` class. I personally like that the API class is light and provides simple CRUD methods.